### PR TITLE
v3.0 changes

### PR DESCRIPTION
**Pull request by judgej**
This replaces 30alpha1-dev, which got hopelessly out of date.

### CODE REVIEW

**Comment by SOMERSOFTAndrewCrook** [comment]
Working to get Omnipay SagePay 3.0 working found that it uses
```php
httpClient->post()
```
 and not 
```php
httpClient->request('POST', ... )
```
as mentioned in [UPGRADE.md](https://github.com/thephpleague/omnipay-common/blob/master/UPGRADE.md). Here is the patch
```patch
diff --git a/src/Message/AbstractRequest.php b/src/Message/AbstractRequest.php
index 7d259e2..d0b7cd2 100644
--- a/src/Message/AbstractRequest.php
+++ b/src/Message/AbstractRequest.php
@@ -335,10 +335,11 @@
 
         $httpResponse = $this
             ->httpClient
-            ->post(
-                $this->getEndpoint(),
-                ['Content-Type' => 'application/x-www-form-urlencoded'],
-                http_build_query($data)
+            ->request(
+                'POST',
+                 $this->getEndpoint(),
+                 ['Content-Type' => 'application/x-www-form-urlencoded'],
+                 http_build_query($data)
             );
 
         // The body is a string.
```

**Comment by judgej** [comment]
Thanks Andrew, I'm going to try to get this kick-started again very soon. Too many distractions on other projects have kept me away from this for far too long.

File: src/Message/ServerNotifyRequest.php [hunk 1]
```diff
@@ -24,7 +25,7 @@ class ServerNotifyRequest extends AbstractRequest implements NotificationInterfa
     /**
      * Initialise the data from the server request.
      */
-    public function __construct(ClientInterface $httpClient, HttpRequest $httpRequest)
+    public function __construct(Client $httpClient, HttpRequest $httpRequest)
```
Comment (barryvdh) [inline src/Message/ServerNotifyRequest.php]: This should actually be ClientInterface now

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,9 @@
 language: php
 
 php:
-  - 5.4
-  - 5.5
   - 5.6
   - 7.0
   - 7.1
```
Comment (barryvdh) [inline .travis.yml]: You might want to install 7.2 also

**Comment by judgej** [comment]
All changes superseded by other changes, as this PR got hopelessly out of date. Thank you all for the updates.

**Pull request closed without merging**